electron/protocol.ts

```diff
@@ -1,29 +1,51 @@
-import { BrowserWindow } from 'electron'
+import { BrowserWindow, dialog, nativeImage } from 'electron'
 import { Game } from './games'
+import i18next from 'i18next'
 
 export async function handleProtocol(window : BrowserWindow, url : string) {
   const [scheme, path] = url.split('://')
   if (!url || scheme !== 'heroic' || !path) {
     return
   }
-  const [command, args_string] = path.split('?')
-  const args = new Map<string, string>()
-  args_string.split(',').forEach((arg) => {
-    const [k, v] = arg.split('=')
-    args.set(k, v)
-  })
+  let [command, arg] = path?.split('/')
+  if (!command || !arg) {
+    command = path
+    arg = null
+  }
   console.log(`ProtocolHandler: received '${url}'`)
   if (command === 'ping') {
-    return console.log('Received ping!', args)
+    return console.log('Received ping! Arg:', arg)
   }
-  const appName = args.get('appName')
-  const game = Game.get(appName)
   if (command === 'launch') {
-    const { is_installed } = await game.getGameInfo()
+    const game = Game.get(arg)
+    const { is_installed, title, art_logo, app_name } = await game.getGameInfo()
     if (!is_installed) {
-      console.log(`ProtocolHandler: "${appName}" not installed, ignoring launch request.`)
+      console.log(`ProtocolHandler: "${arg}" not installed.`)
+      const diag = await dialog.showMessageBox(window, {
+        buttons: [i18next.t('box.yes'), i18next.t('box.no')],
+        cancelId: 1,
+        icon: nativeImage.createFromDataURL(art_logo),
+        message: `${title} ${i18next.t('box.protocol.install.not_installed')}`,
+        title: title
+      })
+      if (diag.response === 0) {
+        window.webContents.send('install', {
```

Comment:
No I think might be confusing what I said, but here you can send the install command only with the appname, and on the frontend, on the GlobalState.tsx file, that line I said above, you need to copy paste how we are handling the launch, but instead call the install command with these arguments, but no on the backend, on the frontend.
Its two different things.
or maybe just call install, appName here and I can handle the frontend later.

Reply:
Yeah, change this line to `...send('install', app_name), and I'll handle the rest.